File: Daily Test/C++/String Reversal - First & Second Half 6.5.2026.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    string s;
    cin >> s;

    int n = s.length();
    int m = n / 2;

    string first = s.substr(0, m);
    reverse(first.begin(), first.end());

    if(n % 2 == 1) {
        char mid = s[m];
        string second = s.substr(m+1);
        reverse(second.begin(), second.end());
        cout << first + mid + second;
    } else {
        string second = s.substr(m);
        reverse(second.begin(), second.end());
        cout << first + second;
    }
}

File: Daily Test/Java/String Reversal - First & Second Half 6.5.2026.java
import java.util.*;

public class Hello {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String s = sc.next();
        int n = s.length();
        int m = n / 2;

        StringBuilder first = new StringBuilder(s.substring(0, m)).reverse();

        if(n % 2 == 1){
            char mid = s.charAt(m);
            StringBuilder second = new StringBuilder(s.substring(m+1)).reverse();
            System.out.println(first + "" + mid + second);
        } else {
            StringBuilder second = new StringBuilder(s.substring(m)).reverse();
            System.out.println(first + "" + second);
        }
    }
}